lib/msal-common/src/network/NetworkManager.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { ServerAuthorizationTokenResponse } from "../server/ServerAuthorizationTokenResponse";
+import { INetworkModule } from "./INetworkModule";
+import { ServerTokenRequestParameters } from "../server/ServerTokenRequestParameters";
+import { ICacheStorage } from "../cache/ICacheStorage";
+import { AccessTokenKey } from "../cache/AccessTokenKey";
+import { RequestThumbprint } from "./RequestThumbprint";
+import { Constants } from "../utils/Constants";
+import { ICrypto } from "../crypto/ICrypto";
+import { TokenExchangeParameters } from "../request/TokenExchangeParameters";
+import { ClientInfo } from "../auth/ClientInfo";
+import { ErrorValue } from "../cache/ErrorValue";
+
+export type NetworkResponse = {
+    headers: Headers;
```

Comment:
I haven't tested this, but should `headers` instead be represented by an something like `Map<string, string>` to accomodate XHR/IE?

Reply:
Yeah, I would not rely on classes that are specific to one of the platforms.

Reply:
We are changing this for `msal-node`

Reply:
@sameerag can you elaborate?